masu/processor/tasks.py

```diff
@@ -69,12 +70,22 @@ def get_report_files(customer_name,
 
     # initiate chained async task
     for report_dict in reports:
-        request = {'schema_name': schema_name,
-                   'report_path': report_dict.get('file'),
-                   'compression': report_dict.get('compression')}
-        LOG.info('Enqueuing processing task for file %s', report_dict.get('file'))
-        result = process_report_file.delay(**request)
-        LOG.info('Processing task enqueued. Task ID: %s', str(result))
+        stats = ReportStatsDBAccessor(report_dict.get('file'), provider_id)
+        last_start = stats.get_last_started_datetime()
+        last_end = stats.get_last_completed_datetime()
+
+        # only queue up a processing job if there's none currently in-progress.
+        if not last_start or not last_end or last_start < last_end:
+            request = {'schema_name': schema_name,
+                       'report_path': report_dict.get('file'),
+                       'compression': report_dict.get('compression')}
+            result = process_report_file.delay(**request)
```

Comment:
I know it's a carry over, but at this point making the request container is superfluous. 

`result = process_report_file.delay(schema_name, report_dict.get('file'), report_dict.get('compression'))` takes less space and is, IMHO, more readable. 

Using `*args` and `**kwargs` is great when you maybe have a function that takes a variable number of args and processes whatever you throw at it. In this case we know the task function has 3 well defined parameters and it's 1 less mental step for a reader to see arguments passed directly into a function, instead of mentally unpacking the dictionary to pass them in. I know in this case the dictionary is right there, so it's really not that much additional effort, but my point is, even that small amount of effort need not be expended.

Reply:
Understood. I've been using Python long enough that `kwargs` is pretty normal to me, so I don't find it to be significant cognitive effort. To me, it's very idiomatic.

I'm fine with changing it, if desired. I just need clear direction on whether this is a blocker or not.

Reply:
I might agree if there were keyword arguments to this function with which to use kwargs. As it stands you *must* be passing in a dictionary with **all** of the arguments to the function or the call will fail. Because "explicit is better than implicity" it makes more sense to clearly call out the arguments. The spirit of kwargs is really around flexibility and optional arguments. Because there are no optional arguments, that flexibility doesn't exist here. The following will fail:

```
request = {'schema_name': schema_name, 'report_path': report_dict.get('file')}
result = process_report_file.delay(**request)
```

I was really just trying to point that out with a comment. This isn't blocking, and it doesn't need to be changed.